### PR TITLE
fix: loadDatabases KEGG download + alternative uniprot ID source

### PR DESCRIPTION
**Pull request by edkerk**
### Main improvements in this PR:
- loadDatabases stopped downloading KEGG
- providing `data/uniprotConversion.tsv` in adapter folder can be used as alternative source of converting model genes (1st column) to uniprot identifiers (2nd column).

**I hereby confirm that I have:**

- [ ] Tested my code with [all requirements](https://github.com/SysBioChalmers/GECKO) for running GECKO
- [ ] Selected `devel` as a target branch (top left drop-down menu)

### CODE REVIEW

File: src/geckomat/model_adapter/ModelAdapter.m [hunk 1]
```diff
@@ -1,41 +1,59 @@
...
         function genes = getUniprotCompatibleGenes(obj,inGenes)
             genes = inGenes;
         end
-        
+
+        function uniprotIDs = getUniprotIDsFromTable(obj,modelGenes)
+            conversionTable = fullfile(obj.params.path,'data','uniprotConversion.tsv');
+            if exist(conversionTable,'file')
+                fID=fopen(conversionTable,'r');
+                conversionTable = textscan(fID,'%q %q','Delimiter','\t','HeaderLines',1);
```
Comment (Yu-sysbio) [inline src/geckomat/model_adapter/ModelAdapter.m:30]: The text in the `uniprotConversion.tsv` file cannot be read.
The resulting `conversionTable` is just 1x2 cell, i.e., `[] []`

**Comment by Yu-sysbio** [comment]
I just followed the protocol in the procedure section but still failed with the warning and errors: 

> Warning: The 1516 gene(s) reported in noUniprot cannot be found in data/uniprot.tsv, these will not be
> enzyme-constrained. If you intend to use different Uniprot data (e.g. from a different proteome identifier), make sure
> you first delete the existing data/uniprot.tsv file. 

> Error using cell/ismember (line 34)
> Input A of class double and input B of class cell must be cell arrays of character vectors, unless one is a character
> vector.
> 
> Error in addMets (line 106)
> I=ismember(metsToAdd.mets,model.mets);
> 
> Error in makeEcModel (line 330)
>     model = addMets(model,proteinMets);

I am not able to identify the bugs and therefore just put my code here for your reproduction.

My code here:

> geckoRoot = findGECKOroot;
> ModelAdapterManager.setDefaultAdapterFromPath(fullfile(geckoRoot,'userData','ecEcoli'));
> modelAdapter = ModelAdapterManager.getDefaultAdapter();
> model = loadConventionalGEM;
> DB = loadDatabases('kegg');
> fID = fopen(fullfile(modelAdapter.params.path,'data','uniprotConversion.tsv'),'w'); 
> output = transpose([DB.kegg.keggGene,DB.kegg.uniprot]);
> fprintf(fID,'%s %s\n','genes', 'uniprot');
> fprintf(fID,'%s %s\n',output{:});
> fclose(fID);
> [ecModel, noUniprot] = makeEcModel(model);

And I attached my adapter here:
[iML1515Adapter.m.zip](https://github.com/SysBioChalmers/GECKO/files/10860216/iML1515Adapter.m.zip)

**Comment by edkerk** [comment]
@Yu-sysbio 
Thanks for providing so much info, the problem is in these two lines (now also corrected in the protocol):

Note the `\t`

```
fprintf(fID,'%s\t%s\n','genes', 'uniprot');
fprintf(fID,'%s\t%s\n',output{:});
```

**Comment by Yu-sysbio** [comment]
Now I passed smoothly the `makeEcModel` step, 
but could not get any EC number by running `ecModel = getECfromDatabase(ecModel)`.
The reason is the gene identifiers in [data.mat.zip](https://github.com/SysBioChalmers/GECKO/files/10877492/data.mat.zip) do not perfectly match model.genes.

**Comment by edkerk** [comment]
Failure of test is known consequence of https://github.com/SysBioChalmers/RAVEN/issues/469.

**Pull request merged**